### PR TITLE
1875 - Add Discord To Welcome Email

### LINKED ISSUE #1875: Discord Invite Email
We should send an email to members that are;

- newly created
- home members of the division
- that we have an email address for

Inviting them to join our Discord server.

### PR DESCRIPTION
**Pull request by atoff**
image

Closes #1875

### CODE REVIEW

File: routes/web-main.php [hunk 1]
```diff
@@ -120,9 +120,7 @@
     'middleware' => 'auth_full_group',
 ], function () {
     Route::get('/')->uses('Registration@show')->name('show');
-    Route::redirect('/invite', config('services.discord.invite_url'))->name('invite');
```
Comment (atoff) [inline routes/web-main.php:123]: @CalumTowers I assumed we don't need these anymore?

File: routes/web-main.php [hunk 2]
```diff
@@ -120,9 +120,7 @@
...
     Route::get('/create')->uses('Registration@create')->name('create');
-    Route::get('/store')->uses('Registration@store')->name('store');
```
Comment (CalumTowers) [inline routes/web-main.php:125]: This is important to keep as it is where the OAuth posts to.

File: tests/Feature/Services/DiscordTest.php [hunk 3]
```diff
@@ -64,33 +64,4 @@ public function testItPassesParamatersToOAuth()
 
         $this->assertEquals($parameters->except('state')->toArray(), $expected);
     }
-
-    /** @test */
-    public function testItRedirectsWhenCodeMissing()
```
Comment (CalumTowers) [inline tests/Feature/Services/DiscordTest.php:69]: This should remain, it is a valid test

File: routes/web-main.php (hunk 1 above)
Comment (CalumTowers) [reply routes/web-main.php:123]: Don't need the invite URL nope!

**Comment by CalumTowers** [comment]
@atoff Looks good - I'd be quite happy to get rid of that entire table telling them their email etc, and just make the email a tad more welcoming (as in, thanks for registering with VATSIM.net, you chose VATSIM UK as your home Division, blah blah)

**Comment by atoff** [comment]
@CalumTowers Tom is happy with this now so should be good to go!

**Comment by CalumTowers** [comment]
@atoff Thanks, I've made some small changes, let me know what you think!

image

File: resources/views/emails/mship/account/welcome.blade.php [hunk 4]
```diff
@@ -2,73 +2,57 @@
...
-    For both Pilot and ATC training our online system will allow you to book mentoring (once you are enrolled as per the above links):
-    {!! link_to("https://cts.vatsim.uk/", "https://cts.vatsim.uk/") !!}
+    We recommend making use of our <a href="{{route('discord.show')}}">Discord</a>, <a href="https://community.vatsim.uk">community forum</a>, or <a href="{{route('site.community.teamspeak')}}">TeamSpeak</a> to get help from members and division staff.
 </p>
-
 <p>
-    The UK TeamSpeak Server and Forum - these are used for voice and written communications for the entire UK community. Feel free to join the community and ask questions and meet others:
-    <li> TeamSpeak - {!! link_to_route('site.community.teamspeak') !!}</li>
-    <li> Forum - {!! link_to("https://community.vatsim.uk/", "https://community.vatsim.uk/") !!}</li>
+    You'll be in good company, although there is lots to learn, there are plenty of people ready to support you.
```
Comment (atoff) [inline resources/views/emails/mship/account/welcome.blade.php:50]: This doesn't sound quite right, maybe:

You'll be in good company, and although...

Apart from that all good here!

Comment (CalumTowers) [reply resources/views/emails/mship/account/welcome.blade.php:50]: Changed it to `Although there is a lot to learn, you will be in good company, surrounded by like-minded simming enthusiasts ready to help you get started.`

**Pull request merged**